lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -1611,9 +1611,37 @@ bool Equivalent(llvm::Optional<T> l, T r) {
   return IMPL();
 #endif
 
+
+#ifndef NDEBUG
+using DAGCheckSet = llvm::SmallPtrSet<swift::Demangle::NodePointer, 8>;
+static bool ContainsCyclesImpl(DAGCheckSet &visited,
```

Comment:
The cycle  check may be a good addition to `swift::Demangle` itself. I'm assuming lldb isn't the only consumer that would want to ensure cycles don't accidentally occur.